init-common/prepare-dvm-sysvinit.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Source Qubes library.
+. /usr/lib/qubes/init-common/functions
+
+possibly_run_save_script
+umount /rw
+dmesg -c >/dev/null
+free | grep Mem: | (read a b c d ; qubesdb-write /qubes-used-mem $c)
+# give dom0 time to read some entries, when done it will shutdown qubesdb,
+# so wait for it
+qubesdb-watch /stop-qubesdb
+# just to make sure
+systemctl stop qubes-db.service
+
+# we're still running in DispVM template
+echo "Waiting for save/restore..."
+# the service will start only after successful restore
+systemctl start qubes-db.service
+echo Back to life.
+
+# Reload random seed
+qubesdb-read /qubes-random-seed | base64 -d > /dev/urandom
+qubesdb-rm /qubes-random-seed
```

Comment:
Actually, it honestly looks like the correct version is the systemd one, and fuck everything else.  The other is not correct and will not work with new dom0s.